R/write_instance_data.R

```diff
@@ -38,20 +39,22 @@ write_instance_data <- function(ncFile, attData, instanceDimName = "instance") {
 	if(charDimLen>0) {
 		char_dim <- ncdim_def('char', '', 1:charDimLen, create_dimvar=FALSE)
 	}
+	col <- 1
 	for(colName in names(attData)) {
 		if(grepl(class(attData[colName][[1]]), "character")) {
-			vars <- c(vars, list(ncvar_def(name=colName, units = "unknown", dim = list(char_dim, instance_dim),
+			vars <- c(vars, list(ncvar_def(name=colName, units = units[col], dim = list(char_dim, instance_dim),
 																		 prec = types[[class(attData[colName][[1]])]])))
 		} else if(grepl(class(attData[colName][[1]]), "integer")) {
-			vars <- c(vars, list(ncvar_def(name=colName, units = "unknown", dim = instance_dim,
+			vars <- c(vars, list(ncvar_def(name=colName, units = units[col], dim = instance_dim,
 																		 prec = types[[class(attData[colName][[1]])]], missval = -9999)))
 		} else {
-			vars <- c(vars, list(ncvar_def(name=colName, units = "unknown", dim = instance_dim,
+			vars <- c(vars, list(ncvar_def(name=colName, units = units[col], dim = instance_dim,
 																		 prec = types[[class(attData[colName][[1]])]], missval = NA)))
 		}
+		col <- col + 1
 	}
 	
-	nc <- nc_create(filename = ncFile, vars = vars)
+	nc <- nc_create(filename = ncFile, vars = vars, force_v4 = force_v4)
```

Comment:
You might want to just pass a `...` through to nc_create so you don't have to define each argument you want to allow.